src/machines/modelingMachine.ts

```diff
@@ -1994,42 +1999,166 @@ export const modelingMachine = setup({
 
         // Extract inputs
         const ast = kclManager.ast
-        const { selection, thickness } = input
+        const { selection, thickness, nodeToEdit } = input
+        let variableName: string | undefined = undefined
+        let insertIndex: number | undefined = undefined
 
-        // Insert the thickness variable if it exists
-        if (
-          'variableName' in thickness &&
-          thickness.variableName &&
-          thickness.insertIndex !== undefined
-        ) {
-          const newBody = [...ast.body]
-          newBody.splice(
-            thickness.insertIndex,
-            0,
-            thickness.variableDeclarationAst
+        // If this is an edit flow, first we're going to remove the old extrusion
+        if (nodeToEdit && typeof nodeToEdit[1][0] === 'number') {
+          // Extract the plane name from the node to edit
+          const variableNode = getNodeFromPath<VariableDeclaration>(
+            ast,
+            nodeToEdit,
+            'VariableDeclaration'
           )
+          if (err(variableNode)) {
+            console.error('Error extracting name')
+          } else {
+            variableName = variableNode.node.declaration.id.name
+          }
+
+          // Removing the old statement
+          const newBody = [...ast.body]
+          newBody.splice(nodeToEdit[1][0], 1)
           ast.body = newBody
+          insertIndex = nodeToEdit[1][0]
+        }
+
+        // Turn the selection into the faces list
+        const clonedAstForGetExtrude = structuredClone(ast)
+        const faces: Expr[] = []
+        let pathToExtrudeNode: PathToNode | undefined = undefined
+        for (const graphSelection of selection.graphSelections) {
```

Comment:
This is the old code from addShell.ts to turn the graph selection into a list of string and tag identifiers for the faces kwarg. It's extremely cumbersome and I hate it.

Reply:
https://github.com/KittyCAD/modeling-app/pull/5525#discussion_r1991946566